packages/components/src/snackbar/index.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * External dependencies
+ */
+import { flow, noop } from 'lodash';
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { RawHTML } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { Button } from '../';
+
+const stopPropagation = ( event ) => event.stopPropagation();
+
+function Snackbar( {
+	className,
+	status,
+	children,
+	actions = [],
+	__unstableHTML,
+	onRemove = noop,
+} ) {
+	const classes = classnames( className, 'components-snackbar', 'is-' + status );
+
+	if ( __unstableHTML ) {
```

Comment:
How long has `__unstableHTML` been around and is this a pattern that should continue to be replicated in new things?  Is there any attempts to remove the necessity for it being __unstable?

Reply:
That's a good question, I think we should do an audit of the experimental APIs and see which ones can be made "stable".

That said, I think `__unstable` is a bit different in the sense that it's not meant to be made stable at all. It's something used internally that third-party developers shouldn't rely on. That doesn't mean the use-case here is not legit and shouldn't be exposed but it shouldn't in its current form. For this particular one @aduth knows more.

Reply:
I think in this case, it was added to support back-compat with the old notice system which allowed plugins to add arbitrary html (and was related to some work @aduth did with automatically porting usage of `add_action( 'admin_notice' )` over to the notices in GB.

Reply:
Maybe we can just remove it, in that case as we don't support these admin notices anymore

Reply:
I think at a minimum it should be removed for the new snackbar notices. I'm not comfortable making the call in general for all notices though.

Reply:
So... this one's been on my mind lately.

It's not used anywhere in Gutenberg currently. It was originally part of some compatibility effort for server-rendered notices which was dropped late in the 5.0 release cycle. In fact, I still have an open pull request to remove it: #12440

This sat longer than I'd liked, and I know of several occurrences of people indicating they're using it, despite repeated warnings of its unstable nature. The last time it came up in a #core-js chat, we said that we didn't care, and that we'd still remove it since we're not obligated to the backwards-compatibility. It doesn't sit well with me though, largely on the point that we let this specific one linger, and thus it's more "on us".

My latest thought was to do a search in the plugins repository for who might be using it, and to both (a) consider their use case and (b) reach out to them directly ahead of its removal.

I ran the search now:

https://wpdirectory.net/search/01DAVD7SA20EYCQM5K456HMYNK

There aren't quite as _many_ hits as I had expected, but certainly some popular ones.

I think regardless whether we decide to do it now (for which an argument could certainly be made), considering the use-cases is valid.

For example, one instance I'd heard that someone still chose to use `__unstableHTML` was the need for dismissed notices to be persistent. This seems like a useful feature that we'd want to consider to implement anyways.

Reply:
Maybe we should flag this in #core-js and also try reaching out to those using it to gather feedback on the use-case requirements for it?  Then we could try to implement something that satisfies common use-cases.

Reply:
> Maybe we should flag this in #core-js and also try reaching out to those using it to gather feedback on the use-case requirements for it? Then we could try to implement something that satisfies common use-cases.

I've added it to the agenda as an item to discuss: 

https://docs.google.com/document/d/1h21-24QSi02QXr3qMqUAkzpT4B02Ep3ZKKI2qhgwUUw/edit